### PR TITLE
version addon images together with the rest of Kubermatic

### PR DESCRIPTION
**Pull request by kdomanski**
**What this PR does / why we need it**:
It will simplify version management of addon images, making it easier to backport some but not all changes to release branches.

```release-note
NONE
```

### CODE REVIEW

File: api/hack/ci/ci-push-images.sh [hunk 1]
```diff
@@ -16,5 +18,31 @@ time make -C api build
...
+    if [[ -z "$TAG" ]]; then
+      continue
+    fi
+
+    if [[ "$TAG" -eq "$GIT_HEAD_HASH" ]]; then
+      continue
+    fi
+
+    echo "Tagging ${TAG}"
+    docker tag quay.io/kubermatic/addons:${GIT_HEAD_HASH} quay.io/kubermatic/api:${TAG}
```
Comment (mrIncompetent) [inline api/hack/ci/ci-push-images.sh]: Should this rather be:
```
docker tag quay.io/kubermatic/addons:${GIT_HEAD_HASH} quay.io/kubermatic/addons:${TAG}
```
?

File: api/hack/ci/ci-push-images.sh [hunk 2]
```diff
@@ -16,5 +18,31 @@ time make -C api build
...
+done
+echodate "Successfully finished building addon image"
+
+echodate "Pushing addon images"
+time docker build -t quay.io/kubermatic/addons:${GIT_HEAD_HASH} ./addons
```
Comment (mrIncompetent) [inline api/hack/ci/ci-push-images.sh]: Why do we built the image twice?

File: api/hack/ci/ci-push-images.sh (hunk 1 above)
Comment (kdomanski) [reply api/hack/ci/ci-push-images.sh]: Fixed.

File: api/hack/ci/ci-push-images.sh (hunk 2 above)
Comment (kdomanski) [reply api/hack/ci/ci-push-images.sh]: Fixed.

File: api/hack/ci/ci-push-images.sh [hunk 3]
```diff
@@ -16,5 +18,30 @@ time make -C api build
 echodate "Successfully finished building binaries"
 
 echodate "Building and pushing quay images"
-retry 5 ./api/hack/push_image.sh $GIT_HEAD_HASH $(git tag -l --points-at HEAD) "latest"
+retry 5 ./api/hack/push_image.sh "${TAGS[@]}"
 echodate "Sucessfully finished building and pushing quay images"
+
+echodate "Building addons"
+time docker build -t quay.io/kubermatic/addons:${GIT_HEAD_HASH} ./addons
```
Comment (alvaroaleman) [inline api/hack/ci/ci-push-images.sh:25]: Can we retry this? Docker sometimes has hiccups

Comment (kdomanski) [reply api/hack/ci/ci-push-images.sh:25]: We don't retry docker builds anywhere else.

Comment (alvaroaleman) [reply api/hack/ci/ci-push-images.sh:25]: Oh interesting, I thought we did that :thinking:

**Comment by alvaroaleman** [comment]
@kdomanski Why is that presubmit failing? Overriding is not really a solution, it will fail for all PRs after this once this got merged
/hold

**Comment by kdomanski** [comment]
The reason it's failing is because the hardcoded image version tag is replaced with the dynamic `__KUBERMATIC_TAG__` matching the other images. After this is PR in, the pre-submit becomes obsolete for every branch that has it.

**Comment by kdomanski** [comment]
@kubermatic-bot ping

**Pull request merged**